Dashboard UI Fixes - SEAB 4712/SEAB 5198/SEAB 4713

KimberleyChong opened the pull request:
**Description**
Mix of Dashboard related UI fixes so let's dive in:

**My-Sidebar:**

Dividing line was missing the purple highlighting when visiting the dashboard page.  The `routerLink` needed to be updated, and cleaning up I removed the `[routerLinkActiveOptions]="{ exact: true }"` since that was needed originally when the dashboard button would link to the homepage (ie. routerLink="" ).
| Before | After |
|:---:|:---:|
| ![image](https://user-images.githubusercontent.com/97123241/219683988-586198a4-5455-4cb7-afc3-d3b2fce6b054.png) | ![image](https://user-images.githubusercontent.com/97123241/219684019-77fb792a-8d17-4d54-9c20-32d56535e250.png) | 

Added the "User Guide" link to the sidebar:
| Before | After |
|:---:|:---:|
| ![image](https://user-images.githubusercontent.com/97123241/219693530-c627cc94-805f-4441-a864-0f7f725fe20d.png) | ![image](https://user-images.githubusercontent.com/97123241/219693692-ec2a203b-e6d8-4972-b031-31f1180597d9.png) | 

**Navbar:**

"My Dockstore" button was using the wrong icon, should be the smaller light purple one:

| Before | After | Mocks
|:---:|:---:|:---:|
| ![image](https://user-images.githubusercontent.com/97123241/219694691-95d6f315-f2ba-48aa-b5b1-435b33459505.png) | ![image](https://user-images.githubusercontent.com/97123241/219694774-ad8a5463-ee5c-4807-8135-401349cfe2b5.png) | ![image](https://user-images.githubusercontent.com/97123241/219695293-48ff4ba3-0280-4dd2-81c3-e21f7c9957d8.png)

Small updates to the new dropdown, bolded text, new "Log out" icon, colour changes and some additional padding.

| Before | After | Mocks
|:---:|:---:|:---:|
| ![image](https://user-images.githubusercontent.com/97123241/219695600-8a995baf-849a-4526-9681-d9b63b126cfd.png) | ![image](https://user-images.githubusercontent.com/97123241/219695521-25560fbd-c25a-46c6-9ce5-17f7f386dcaf.png) | ![image](https://user-images.githubusercontent.com/97123241/219695687-36d261c4-56b1-40b6-96bc-89dbb9e2822b.png)

Looking at it now, there is no line in the mocks, so this `mat-divider` should be removed:
https://github.com/dockstore/dockstore-ui2/blob/385cce574745a7d079138a907900750be6f06650/src/app/navbar/navbar.component.html#L105

I'll wait for any additional feedback before pushing anything.

**Organization Box:**

Pending request/invitation icons should be greyed out if equal to zero.
| Before | After |
|:---:|:---:|
| ![image](https://user-images.githubusercontent.com/97123241/219698417-b265109d-7c6c-473d-97a2-28d72b2c699f.png) | ![image](https://user-images.githubusercontent.com/97123241/219698454-a9c1d156-50ee-4c26-ba27-493ef2cec0cb.png) |

Noticed pending requests/invitations weren't updating so I changed them to `observables`, updating as expected now.
https://github.com/dockstore/dockstore-ui2/blob/385cce574745a7d079138a907900750be6f06650/src/app/home-page/widget/organization-box/organization-box.component.ts#L23-L24
https://github.com/dockstore/dockstore-ui2/blob/385cce574745a7d079138a907900750be6f06650/src/app/home-page/widget/organization-box/organization-box.component.ts#L42-L43
![image](https://user-images.githubusercontent.com/97123241/219700352-c40c8862-9c89-4916-b5be-dc90f827ea35.png)

Empty card was updated with the circle org icon:
| Before | After |
|:---:|:---:|
| ![image](https://user-images.githubusercontent.com/97123241/219706140-47164bcf-9796-4846-80ab-950044a45a7a.png) | ![image](https://user-images.githubusercontent.com/97123241/219706634-68d9d5eb-4b0b-4e77-8ac9-aeb9be8185df.png) | 

Added the search bar (similar implementation to `entry-box.component`), changed "Recent Activity" to "My Organizations", and switched from using the organization `Name` to the `DisplayName`.

| Before | After |
|:---:|:---:|
| ![image](https://user-images.githubusercontent.com/97123241/219702783-bbb3400e-9586-4200-9dd1-cb6e113c8f16.png) | ![image](https://user-images.githubusercontent.com/97123241/219702681-ae6d332e-fdf8-407f-9b0f-38692d1ed429.png) |

Changes to the Org box still aren't complete, we have a ticket to add a new API that returns a user's organization events (SEAB-5266) and a follow-up ticket to use that API and display a user's organization events following the mocks (SEAB-5267).

Last change, the pending requests/invitations links were updated to go to the "Requests" tab of the account page:
`routerLink="/accounts" [queryParams]="{ tab: 'requests' }"`

Same for the link on this organization pop-up which previously linked to the default tab on the accounts page:
![image](https://user-images.githubusercontent.com/97123241/219705798-ac714cd3-71c5-4c31-a269-8d40acbb0597.png)

Finally, just an overview of the current state of the Dashboard:
![image](https://user-images.githubusercontent.com/97123241/219707738-ba95c34a-84f0-466d-9dd8-04e54a52b2a7.png)

**Review Instructions**


**Issue**
[SEAB-4712](https://ucsc-cgl.atlassian.net/browse/SEAB-4712) 
[SEAB-5198](https://ucsc-cgl.atlassian.net/browse/SEAB-5198)
[SEAB-4713](https://ucsc-cgl.atlassian.net/browse/SEAB-4713)

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.
- [x] Check whether this PR disables tests. If it legitimately needs to disable a test, create a new ticket to re-enable it in a specific milestone. 


[SEAB-4712]: https://ucsc-cgl.atlassian.net/browse/SEAB-4712?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

coverbeck commented on src/app/home-page/widget/organization-box/organization-box.component.ts:
```diff
@@ -38,45 +39,22 @@ export class OrganizationBoxComponent extends Base implements OnInit {
-            }
+          myOrgs.forEach(() => {
+            this.listOfOrganizations = myOrgs.slice(0, 7);
+            this.totalOrgs = myOrgs.length;
```
This is going to be the total of filtered organizations you belong to, not the total organizations you belong to. So if belong to at least one org, but type in a filter that doesn't match any of your orgs, you're going to get the message `You are not in any organizations`, which seems wrong.

coverbeck commented on src/app/home-page/widget/organization-box/organization-box.component.html:
```diff
@@ -42,22 +42,31 @@
+          <img src="../../../../assets/svg/organization-circle.svg" alt="my organizations" class="widget-box-type-logo mb-4" />
           <span>You are not in any organizations </span>
           <div class="size-small text-align-center">
             <span>Use the </span>
```
For line 67/76, I can't comment there, and it was already like this, but my IDE is flagging the id `requestOrganizationButton` as a duplicate id. Looks like the id isn't referenced anywhere, so I think you can just remove both ids.

coverbeck commented on src/app/home-page/widget/organization-box/organization-box.component.ts:
```diff
@@ -38,45 +39,22 @@ export class OrganizationBoxComponent extends Base implements OnInit {
-            if (this.firstCall) {
-              this.totalOrgs += 1;
-            }
+          myOrgs.forEach(() => {
```
You don't need the `forEach()` at all, the body currently just repeats the same operation each time.

KimberleyChong replied:
Yeah no I agree the `forEach()` is redundant, I was following the other dashboard boxes for consistency and couldn't understand why the `forEach()` was there...:
https://github.com/dockstore/dockstore-ui2/blob/b7ddce0c20b7c4991750cfccd039dbe225a91d74/src/app/home-page/widget/entry-box/entry-box.component.ts#L93-L103

I couldn't explain why at the time but when I removed the `forEach()` the search box would go blank when no results returned (From your previous comment I see its because `this.totalOrgs` is set to zero!!):

https://user-images.githubusercontent.com/97123241/219804286-6c4d4cd1-7ee7-4703-a0a4-8d59fa8a4395.mp4

However, with the `forEach()` this is avoided somehow... now I see that it's because the `forEach()` never runs (because `myOrgs` is null) and so the previous state of `this.listOfOrganizations` and `this.totalOrgs` is never reassigned.

https://user-images.githubusercontent.com/97123241/219804487-cd28d349-e7eb-4860-a597-197285c5cc44.mp4

I'll remove the `forEach()`, then addressing your previous comment, assigning `this.totalOrgs` correctly will prevent the card from going to `You are not in any organizations`.

coverbeck commented on src/app/home-page/widget/organization-box/organization-box.component.ts:
```diff
@@ -38,45 +39,22 @@ export class OrganizationBoxComponent extends Base implements OnInit {
+            this.totalOrgs = myOrgs.length;
           });
         },
         (error: HttpErrorResponse) => {
```
It was like this, but you could set `this.listOfOrganizations = []` in the error block to handle this case:

1. User's organizations are successfully fetched
2. User types in some text and there's an error fetching the organizations, e.g., the network connection is lost, web service fails, etc.
3. We display the snackbar message, but the organizations from step are still displayed.

You could argue it's more user-friendly to keep the old orgs displayed, and maybe it is. On the other hand, our error messages aren't very prominent and the user might be surprised to see orgs that don't match the filter. So your call :)

KimberleyChong commented:
Updates to the `organization-box.component` following Charles' feedback: `totalOrgs` remains constant,  removed `forEach()`, added `this.listOfOrganizations = [];` in the error block, and added a "No matching search results." text when returned result is empty.

Quick demo of the new behaviour:


https://user-images.githubusercontent.com/97123241/220383586-f7864e86-949e-4bd8-a348-7d10616a465a.mp4

Also made a followup ticket SEAB-5268 to update how search is handled in the `entry-box.component` to match.

coverbeck commented on src/app/home-page/widget/organization-box/organization-box.component.html:
```diff
@@ -52,6 +51,7 @@
           />
         </mat-form-field>
         <h5 class="weight-bold mt-0">My Organizations</h5>
+        <div *ngIf="listOfOrganizations.length === 0" class="text-align-center date-display mt-5">No matching search results.</div>
```
I think his is going to "flash"; where when the page first loads, it will display this message, the request will complete, then this will go away. It would be nice if it didn't.

KimberleyChong replied:
Adding `&& !isLoading` to the `*ngIf` which avoids the problem, good detail thanks. Making a note to update the other `entry-boxes`

coverbeck commented on src/app/home-page/widget/organization-box/organization-box.component.ts:
```diff
@@ -19,10 +20,11 @@ import { AlertService } from 'app/shared/alert/state/alert.service';
   totalOrgs: number = 0;
-  pendingRequests: Array<OrganizationUser> = [];
-  pendingInvites: Array<OrganizationUser> = [];
+  firstCall: Boolean = true;
```
https://stackoverflow.com/questions/64443288/what-is-the-difference-between-boolean-and-boolean-in-typescript -- you should use `boolean` instead.

KimberleyChong replied:
Ah neat, didn't know that thanks

denis-yuen commented on src/app/my-sidebar/my-sidebar.component.html:
```diff
@@ -72,9 +65,12 @@
   </button>
   <hr routerLink="/accounts" routerLinkActive="active" />
 
+  <!-- User Guide -->
```
Think docs.dockstore.org may be more of a user guide, the quick start is really just a quick-start. Since we have docs up in the header, think this should say "Quickstart"

denis-yuen reviewed (approved):
> I'll wait for any additional feedback before pushing anything.

 looking good so far

> Finally, just an overview of the current state of the Dashboard:
 
Looks good, and thanks for catching all the details!

KimberleyChong commented:
Last change: "User Guide" updated to "Quickstart":

![image](https://user-images.githubusercontent.com/97123241/220655643-068101b3-584e-4a7e-b7b7-5f0e6e832358.png)
